cmd/skywire-cli/commands/config/gen.go

```diff
@@ -203,46 +450,109 @@ var genConfigCmd = &cobra.Command{
 					}
 					owner, err := script.Exec(`stat -c '%U' ` + confPath1).String()
 					if err != nil {
-						logger.Error("cannot stat: " + confPath1)
+						log.Error("cannot stat: " + confPath1)
 					}
 					rootOwner, err := script.Exec(`stat -c '%U' /root`).String()
 					if err != nil {
-						logger.Error("cannot stat: /root")
+						log.Error("cannot stat: /root")
 					}
 					if (owner != rootOwner) && isRoot {
-						logger.Warn("writing config as root to directory not owned by root")
+						log.Warn("writing config as root to directory not owned by root")
 					}
 					if !isRoot && (owner == rootOwner) {
-						logger.Fatal("Insufficient permissions to write to the specified path")
+						log.Fatal("Insufficient permissions to write to the specified path")
 					}
 				}
 			}
 		}
 	},
 	Run: func(cmd *cobra.Command, args []string) {
-		mLog := logging.NewMasterLogger()
-		mLog.SetLevel(logrus.InfoLevel)
-		if serviceConfURL == "" {
-			serviceConfURL = svcConf
-		}
-		//use test deployment
-		if isTestEnv {
-			serviceConfURL = testConf
+
+		log := logger
+
+		if !noFetch {
+			// set default service conf url if none is specified
+			if serviceConfURL == "" {
+				serviceConfURL = utilenv.ServiceConfAddr
+			}
+			//use test deployment
+			if serviceConfURL == "" && isTestEnv {
+				serviceConfURL = utilenv.TestServiceConfAddr
+			}
+			// enable errors from service conf fetch from the combination of these flags
+			wasStdout := isStdout
+			if isStdout && isHide {
```

Comment:
Overriding a flag here silently is not a good option imo. In other places we return error that flags are incompatible in that case.

Reply:
so this was intentional ; marking as resolved